File: internal/model/tile.go
package model

// NoTreasure is when tile has no treasure in it.
const NoTreasure = '.'

// Shape reprents a tile shape.
type Shape int

const (
	// ShapeI represents an I shape.
	ShapeI Shape = iota

	// ShapeT represents a T shape.
	ShapeT Shape = iota

	// ShapeV represents a V shape.
	ShapeV Shape = iota
)

// Tile represents a tile.
type Tile struct {

	// Shape is the tile shape.
	Shape Shape

	// Treasure is the optional tile treasure if present, '.' otherwise.
	Treasure rune
}

File: mobile/src/model/Tile.ts
export enum Shape {
    ShapeI = 0,
    ShapeT = 1,
    ShapeV = 1,
};

export enum Rotation {
    Rotation0 = 0,
    Rotation90 = 90,
    Rotation180 = 180,
    Rotation270 = 270,
};

export interface Tile {
    treasure: string,
    shape: Shape,
};